### ISSUE TITLE
PeerApi: Prove a method to "sync" with the currently-connected peers

### ISSUE THREAD

**Issue by kim** [radicle-dev/radicle-link#208]
This would require protocol support for efficiently computing the intersection
of URNs two peers have locally. As a first step, however, we can just implement
it as a loop which calls `Storage::fetch_repo` for each local URN and connected
peer, ignoring errors.

**Comment by FintanH** [comment]
So interestingly enough @xla pretty much implemented this in proxy. This is likely one of the components that we should migrate back to link.

**Comment by kim** [comment]
Yes would be good, so we can replace it with a more efficient implementation.

**Comment by xla** [comment]
> Yes would be good, so we can replace it with a more efficient implementation.

:100:
